wurst/data/BufferTests.wurst

```diff
@@ -0,0 +1,129 @@
+package BufferTests
+
+import Buffer
+import LinkedList
+
+class SerializableElement implements Serializable
+    int id
+	int value
+	
+	construct()
+
+	construct(Buffer buffer)
+		deserialize(buffer)
+
+	function serialize(Buffer buffer)
+		buffer.writeInt(id)
+		buffer.writeInt(value)
+
+	function deserialize(Buffer buffer)
+		id = buffer.readInt()
+		value = buffer.readInt()
+
+class TestSerializable implements Serializable
+    var name = "SomeName"
+    var id = 0
+    var facing = 3.14
+	var elements = new LinkedList<SerializableElement>
+
+	construct()
+
+	construct(Buffer buffer)
+		deserialize(buffer)
+	
+	function serialize(Buffer buffer)
+		buffer.writeString(name)
+		buffer.writeInt(id)
+		buffer.writeReal(facing)
+		buffer.writeInt(elements.size())
+
+		for element in elements
+			buffer.write(element)
+
+	function deserialize(Buffer buffer)
+		name = buffer.readString()
+		id = buffer.readInt()
+		facing = buffer.readReal()
+		let size = buffer.readInt()
+
+		for i = 0 to size - 1
+			elements.add(new SerializableElement(buffer))
+
+@test
+function stringBufferShortSerializationTest()
+	let sBuffer = new StringBuffer(4)
+	Buffer buffer = sBuffer
+
+	buffer.writeInt(1)
+	buffer.writeInt(2)
+	buffer.writeReal(3.14)
+	buffer.writeInt(3)
+	buffer.writeBoolean(true)
+	buffer.writeBoolean(false)
+	buffer.writeString("a b c d e f g h lol")
+	sBuffer.flush()
+
+	buffer.readInt().assertEquals(1)
+	buffer.readInt().assertEquals(2)
+	buffer.readReal().assertEquals(3.14)
+	buffer.readInt().assertEquals(3)
+	buffer.readBoolean().assertEquals(true)
+	buffer.readBoolean().assertEquals(false)
+	buffer.readString().assertEquals("a b c d e f g h lol")
+
+@test
+function stringBufferLongSerializationTest()
+	let buffer = new StringBuffer(512)
+
+	for i = 0 to 255
+		buffer.writeInt(i)
+		buffer.writeReal(i + 0.125)
+		buffer.writeBoolean(i div 2 == 0)
+		buffer.writeString(i.toString())
+
+	buffer.flush()
+
+	for i = 0 to 255
+		buffer.readInt().assertEquals(i)
+		buffer.readReal().assertEquals(i + 0.125)
+		buffer.readBoolean().assertEquals(i div 2 == 0)
+		buffer.readString().assertEquals(i.toString())
+
+// disabled while there are some bugs preventing it from working correctly (related to the compiler)
```

Comment:
what's the status on this? should be fixed?

Reply:
Works correctly now, re-enabled test